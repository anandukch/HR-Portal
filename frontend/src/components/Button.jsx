/* eslint-disable react/prop-types */
/* eslint-disable no-unused-vars */
import React from "react";
export const Button = ({ text, onClickHandler, className,styles }) => {
    return (
        <button className={className} data-testid="btn-test-id" onClick={onClickHandler} style={styles}>
            {text}
        </button>
    );
};

export default Button;