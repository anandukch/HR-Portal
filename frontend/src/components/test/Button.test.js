/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Button from "../Button";
describe("Check if Button works properly", () => {
    test("Check if the button is rendered", () => {
        const { getByTestId } = render(<Button text="Click" />);
        const element = getByTestId("btn-test-id");
        expect(element).toBeTruthy();
    });
    test("Check if the button text is rendered", () => {
        const text = "Click";
        const { getByText } = render(<Button text={text} />);
        const element = getByText(text);
        expect(element.textContent).toBe(text);
    });

    test("check if onclick is triggered", () => {
        const onClick = jest.fn();
        const { getByTestId } = render(<Button onClickHandler={onClick} />);
        const element = getByTestId("btn-test-id");
        fireEvent.click(element);

        expect(onClick).toHaveBeenCalledTimes(1);
    });

    test("Check if snapshots are mapped properly", () => {
        const onClick = jest.fn();
        const text = "Click";
        const { asFragment } = render(<Button text={text} onClickHandler={onClick} />);
        expect(asFragment()).toMatchSnapshot()
        
    });
});
