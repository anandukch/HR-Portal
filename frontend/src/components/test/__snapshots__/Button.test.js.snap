// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Button works properly Check if snapshots are mapped properly 1`] = `
<DocumentFragment>
  <button
    data-testid="btn-test-id"
  >
    Click
  </button>
</DocumentFragment>
`;
